File: frontend/src/api/provider.ts
import request from './request'

export interface Provider {
  id: number
  code: string
  name: string
  baseUrl: string
  apiKey: string
  status: number
  description?: string
  createTime: string
  updateTime: string
}

export const getProviders = () => {
  return request.get<any, Provider[]>('/api/admin/providers')
}

export const createProvider = (data: Partial<Provider>) => {
  return request.post<any, Provider>('/api/admin/providers', data)
}

export const updateProvider = (id: number, data: Partial<Provider>) => {
  return request.put<any, boolean>(`/api/admin/providers/${id}`, data)
}

export const deleteProvider = (id: number) => {
  return request.delete<any, boolean>(`/api/admin/providers/${id}`)
}

File: gateway-admin/src/main/java/com/aigateway/admin/entity/Provider.java
package com.aigateway.admin.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("provider")
public class Provider {

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 唯一标识，如 openai / deepseek / volcano */
    private String code;

    /** 展示名称 */
    private String name;

    /** 官方 API Base URL */
    private String baseUrl;

    /** API Key */
    private String apiKey;

    /** 1=启用 0=禁用 */
    private Integer status;

    private String description;

    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}

